qiskit/visualization/pass_manager_visualization.py

```diff
@@ -91,66 +90,168 @@ def pass_manager_drawer(pass_manager, filename=None, style=None, raw=False):
     prev_node = None
 
     for index, controller_group in enumerate(passes):
+        subgraph, component_id, prev_node = draw_subgraph(
+            controller_group, component_id, style, prev_node, index
+        )
+        graph.add_subgraph(subgraph)
+
+    output = make_output(graph, raw, filename)
+    return output
+
+
+def _get_node_color(pss, style):
+    # look in the user provided dict first
+    for typ, color in style.items():
+        if isinstance(pss, typ):
+            return color
+
+    # failing that, look in the default
+    for typ, color in DEFAULT_STYLE.items():
+        if isinstance(pss, typ):
+            return color
+
+    return "black"
+
+
+@_optionals.HAS_GRAPHVIZ.require_in_call
+@_optionals.HAS_PYDOT.require_in_call
+def staged_pass_manager_drawer(pass_manager, filename=None, style=None, raw=False):
+    """
+    Draws the staged pass manager.
+
+        This function needs `pydot <https://github.com/erocarrera/pydot>`, which in turn needs
```

Comment:
I know this was just copied and pasted from the `pass_manager_drawer` docstring, but this won't render in sphinx correctly. So don't worry about fixing it here, we can do that in a followup and fix both this and `pass_manager_drawer` (which looking at https://qiskit.org/documentation/stubs/qiskit.visualization.pass_manager_drawer.html definitely isn't right)

```suggestion
        This function needs `pydot <https://github.com/erocarrera/pydot>`__, which in turn needs
```